src/Widgets/WalkLevelQuiz/WalkLevelQuiz.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useRef, useState } from "react";
+import { StyleSheet, Text, View, ScrollView, Modal, TouchableOpacity, SafeAreaView, Platform } from "react-native";
+import InputField, { InputRangeField } from "../../TypesAndInterfaces/config-sync/input-config-sync/inputField";
+import { EDIT_PROFILE_FIELDS_ADMIN, walkLevelMultiplier, walkLevelOptions } from "../../TypesAndInterfaces/config-sync/input-config-sync/profile-field-config";
+import ToastQueueManager from "../../utilities/ToastQueueManager";
+import axios, { AxiosError, AxiosResponse } from "axios";
+import { DOMAIN } from "@env";
+import { useAppDispatch, useAppSelector } from "../../TypesAndInterfaces/hooks";
+import { RootState, updateProfile } from "../../redux-store";
+import { ServerErrorResponse } from "../../TypesAndInterfaces/config-sync/api-type-sync/utility-types";
+import { BackButton, Raised_Button } from "../../widgets";
+import { RootSiblingParent } from 'react-native-root-siblings';
+import theme, { COLORS } from "../../theme";
+import { FormInput } from "../FormInput/FormInput";
+import { FormSubmit } from "../FormInput/form-input-types";
+
+const WalkLevelQuiz = (props:{callback?:((val:number) => void)}):JSX.Element => {
+
+    const dispatch = useAppDispatch();
+    const jwt = useAppSelector((state: RootState) => state.account.jwt);
+    const userID = useAppSelector((state: RootState) => state.account.userID);
+    const userProfile = useAppSelector((state: RootState) => state.account.userProfile);
+    const [walkLevelValue, setWalkLevelValue] = useState(Math.floor(userProfile.walkLevel / 2));
+
+    const onSaveWalkLevel = () => {
+        //@ts-ignore
+        const walkLevel = walkLevelValue * walkLevelMultiplier;
```

Comment:
Add `number` type to both values and typescript should be happy.
![Screenshot 2024-11-11 214643](https://github.com/user-attachments/assets/b74a9464-a800-4186-a30f-fbad3f7b64f6)